### ISSUE TITLE
Select2Multiple adds empty choice

### ISSUE THREAD

**Issue by beruic** [yourlabs/django-autocomplete-light#825]
Whenever I try to use `Select2Multiple` It adds the empty choice to my choices, which doesn't make sense because it is a multi select.

**Comment by beruic** [comment]
Seems like the issue is that the widget adds the empty choice if `self.placeholder` is set. This is done in `WidgetMixin.render_options`. The lines
```
        else:
            if self.placeholder:
                self.choices.insert(0, (None, ""))
```
seems to do it. I guess  the test could be expanded to test for `self.multi =0 False` too.

**Comment by jpic** [comment]
Is this the else for "if self.url" ?

I suppose you're doing a list autocomplete ? I wonder, more details about
your situation could be interresting.

What's self.multi ?

**Comment by beruic** [comment]
Sorry, I forgot a file reference. It's in `src/dal/widgets.py`. Here is a direct link: https://github.com/yourlabs/django-autocomplete-light/blob/22a107df46860d666ff5067e1465c3f67363cc5b/src/dal/widgets.py#L120.

**Comment by beruic** [comment]
Oh, and I am using a `Select2Multiple`, and `self.multi` does not seem to be defined. Did you mean `self.allow_multiple_selected`? That one is `True`.

The empty option is sometimes selected somehow, and really gets in the way for my form treatment.

**Comment by jpic** [comment]
I totally agree, I'm trying to figure out why this hack is there in the first place, and what does it break if we remove it.

**Comment by beruic** [comment]
Cool. Otherwise, I think my PR will fix it.

**Comment by jpic** [comment]
I'm really not sure, because usually tests pass on django  up to 1.10 and in your PR there is no django version that passes tests ...

**Comment by beruic** [comment]
Oh? I admit that I just fixed it in the browser, but I cannot see how this breaks a lot of things. Perhaps the tests themselves are wrong?

**Comment by jpic** [comment]
The tests are mostly (90%) automated selenium tests for the test apps in the django admin.

**Comment by jpic** [comment]
The tests are mostly (90%) automated selenium tests for the test apps in the django admin, so they don't really care about changes at this level.

**Comment by beruic** [comment]
It doesn't seem like it is my change that made it fail. If you look at https://travis-ci.org/yourlabs/django-autocomplete-light/builds, master seems to have the same issues.

**Comment by jpic** [comment]
This sounds like good news, sorry your pr will have to wait until the tests are fixed ... i wish i had more time to maintain this code ^^

**Comment by jpic** [comment]
It seems like the empty choice is required for the placeholder in select2:

https://github.com/yourlabs/django-autocomplete-light/blob/master/test_project/tests/test_widgets.py#L61

**Comment by beruic** [comment]
According to the documentation of Select2, it is only necessary for non-multi value selects. See the Placeholders section on http://select2.github.io/select2/.

**Comment by jpic** [comment]
Perfect, thanks for the heads up, I'm almost there myself, after chipping in ~30 hours of hacking already lol

**Comment by porksmash** [comment]
I am experiencing this issue without using a placeholder on a Select2Multiple widget. Something is generating a default option `<option value selected>---------</option>`, which select2 is automatically adding along with the first item you happen to click in the autocomplete list. The option being there doesn't appear to be the problem - just the presence of the `selected` attribute. 

This option is in the server-generated HTML before any javascript executes, but I have not able to narrow it down just yet. Will update when I do.

**Comment by jpic** [comment]
Feel free to fix it locally, it'll be released as soon as we've checked this hasn't broke any test after the PR's rebase on master.

**Comment by MiltosD** [comment]
How can you save Select2Multiple field and not get error like:
![image](https://user-images.githubusercontent.com/10659857/39923003-8633aabe-552a-11e8-9a1c-cf5ed6278847.png)

I use xhr to get values from a list..
